iep-011/README.adoc

```diff
@@ -0,0 +1,117 @@
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+= IEP-11: 
+
+:toc:
+
+.Metadata
+[cols="2"]
+|===
+| IEP
+| 11
+
+| Title
+| Download packages from Azure
+
+| Author
+| link:https://github.com/olblak[Olivier Vernin]
+
+| Status
+| :speech_balloon: In-process
+
+| Type
+| Architecture
+
+| Created
+| 2018-06-01
+|===
+
+
+
+== Abstract
+link:https://pkg.jenkins.io[pkg.jenkins.io] is a service that distributes Jenkins packages for various Operating Systems.
+The service is currently running on dedicated machines on AWS and rely on third party link:http://mirrors.jenkins.io/status.html[mirrors] to distribute packages.
+The goal here is to identify how we can leverage the usage of our cloud environment(Azure) to have a full control on packages distribution.
+
+== Specification
+=== Data
+Artifacts must be store on Azure container link:https://github.com/jenkins-infra/azure/blob/master/plans/releases-storage.tf[storage] and must host packages for following distributions:
+
+* Debian/Ubuntu
+* Redhat
+* OpenSuse
+* Windows
+* Mac OSX
+* Java Package (WAR)
+
+The way packages are upload on those azure container storage still need to be adressed in another IEP document.
+
+=== Endpoint
+The service will listen on two different endpoints:
+
+* 'pkg.jenkins.io' which accept HTTP/HTTPS requests
+* 'download.jenkins.io' which always redirect HTTP to HTTPS
+
+'download.jenkins.io' will be the new default endpoint and for people who explicitly need a HTTP endpoint, pkg.jenkins.io will be provided as a fallback.
+
+=== Service
+The service will be a web proxy deployed on Kubernetes and it will provide packages for all supported distributions.
+
+.Kubernetes Service Schema
+----
+  +----------------------------+
+  | Ingress:                   |
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  | /debian              +---------->| Service: Debian    +------->| Deployment: Debian     |
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  | /redhat              +---------->| Service: RedHat    +------->| Deployment: RedHat     |
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  | /opensuse            +---------->| Service: OpenSuse  +------->| Deployment: Opensuse   |
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  +----------------------------+
+----
+
+IMPORTANT: As we don't want to rely on local mirrors anymore, we must take into account the user experience when downloading packages from pkg.jenkins.io.
+
+== Motivation
+The motivations are multiple:
+
+1. Do not rely on third party link:http://mirrors.jenkins.io/status.html[mirrors].
```

Comment:
As maintainer of one of the biggest mirrors in Europe, Should I shutdown the server and thus save my own money and my time? /cc @rtyler